Source/Core/CheckoutAPIService.swift

```diff
@@ -5,31 +5,72 @@
 //  Created by Harry Brown on 03/02/2022.
 //
 
+import Foundation
+import UIKit
 import Checkout
+import CheckoutEventLoggerKit
 
 protocol CheckoutAPIProtocol {
-  func createToken(_ paymentSource: PaymentSource, completion: @escaping (Result<TokenDetails, TokenisationError.TokenRequest>) -> Void)
-  var cardValidator: CardValidating { get }
+    func createToken(_ paymentSource: PaymentSource, completion: @escaping (Result<TokenDetails, TokenisationError.TokenRequest>) -> Void)
+    var cardValidator: CardValidating { get }
+    var logger: FramesEventLogging { get }
 }
 
 public class CheckoutAPIService: CheckoutAPIProtocol {
 
     private let checkoutAPIService: Checkout.CheckoutAPIProtocol
     let cardValidator: CardValidating
+    let logger: FramesEventLogging
 
-    public convenience init(publicKey: String, environment: Checkout.Environment) {
-        let checkoutAPIService = Checkout.CheckoutAPIService(publicKey: publicKey, environment: environment)
-        let cardValidator = CardValidator(environment: environment)
+    public convenience init(publicKey: String, environment: Environment) {
+        let checkoutAPIService = Checkout.CheckoutAPIService(publicKey: publicKey, environment: environment.checkoutEnvironment)
+        let cardValidator = CardValidator(environment: environment.checkoutEnvironment)
+        let logger = Self.buildFramesEventLogger(environment: environment, getCorrelationID: { checkoutAPIService.correlationID })
 
-        self.init(checkoutAPIService: checkoutAPIService, cardValidator: cardValidator)
+        self.init(checkoutAPIService: checkoutAPIService, cardValidator: cardValidator, logger: logger)
     }
 
-    init(checkoutAPIService: Checkout.CheckoutAPIProtocol, cardValidator: CardValidating) {
+    init(checkoutAPIService: Checkout.CheckoutAPIProtocol, cardValidator: CardValidating, logger: FramesEventLogger) {
         self.checkoutAPIService = checkoutAPIService
         self.cardValidator = cardValidator
+        self.logger = logger
     }
 
     public func createToken(_ paymentSource: PaymentSource, completion: @escaping (Result<TokenDetails, TokenisationError.TokenRequest>) -> Void) {
         checkoutAPIService.createToken(paymentSource, completion: completion)
     }
+
+    static func buildFramesEventLogger(environment: Environment, getCorrelationID: @escaping () -> String) -> FramesEventLogger {
```

Comment:
Agree with @ehab-al-cko. I would plus that it may be . At this stage cleanest syntax would be if the logic was inside the initialiser for `FramesEventLogger`, so the service initialiser ends up looking like.
```
let logger = FramesEventLogger(environment: environment, getCorrelationID: { checkoutAPIService.correlationID })
```

This can also help us remove the import of UIKit into a file that is called service. Whilst not a huge problem in itself, that import should be an indication that separation may be improved :) .